bundle/config/validate/job_cluster_key_defined.go

```diff
@@ -0,0 +1,66 @@
+package validate
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/libs/diag"
+)
+
+func JobClusterKeyDefined() bundle.Mutator {
+	return &jobClusterKeyDefined{}
+}
+
+type jobClusterKeyDefined struct {
+}
+
+func (v *jobClusterKeyDefined) Name() string {
+	return "validate:job_cluster_key_defined"
+}
+
+func (v *jobClusterKeyDefined) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
+	// Collect all job_cluster_key references from defined tasks
+	jobClusterKeys := make(map[string]location)
+	for i, job := range b.Config.Resources.Jobs {
+		for j, task := range job.Tasks {
+			if task.JobClusterKey != "" {
+				jobClusterKeys[task.JobClusterKey] = location{
+					path: fmt.Sprintf("resources.jobs.%s.tasks[%d].job_cluster_key", i, j),
+					b:    b,
+				}
+			}
+		}
+	}
+
+	if len(jobClusterKeys) == 0 {
+		return nil
+	}
+
+	diags := diag.Diagnostics{}
+
+	// Check if all job_cluster_keys are defined
+	for key, loc := range jobClusterKeys {
+		if !isJobClusterKeyDefined(key, b) {
```

Comment:
The job cluster must be defined in the same job. If it is defined in another job is it not valid.